### PR TITLE
router: Remove the HeaderMatcher value and regex fields.

### PR DESCRIPTION
**Pull request by kavyako**
*Description*: Remove the HeaderMatcher value and regex fields. They are superseded by header_match_specifier oneof exact_match, regex_match.
*Risk Level*: Low
*Testing*: Ran all tests after removing the value and regex fields.
*Docs Changes*: Already up to date.
*Release Notes*: TBD
*Fixes* Issue [2904](https://github.com/envoyproxy/envoy/issues/2904)

### CODE REVIEW

**Comment by mattklein123** [comment]
@kavyako thanks, I had actually started on doing this but didn't finish it. @junr03 can you handle review on this one? I think the big question vs. my other branch is whether to make the oneof required or not.

**Comment by junr03** [comment]
opinion about the open question:

I think the proto is a somewhat confusing without the `required` because then not specifying a `header_matcher_specifier` and specifying `present_match` leads to the same outcome. So my vote is to make the specifier required.

However, making the specifier required is a breaking change, because people might have configs with no specifier set. So we would have to make it required in a new deprecation cycle, right?

**Comment by mattklein123** [comment]
> However, making the specifier required is a breaking change, because people might have configs with no specifier set. So we would have to make it required in a new deprecation cycle, right?

Yeah I was concerned about this also. It's probably not worth it. I would just make sure the default behavior if nothing is specified is very clear in the docs.

**Comment by kavyako** [comment]
Agree with retaining the current behavior and not make `header_match_specifier` a `required` field.

The existing comments above the `oneof` are not showing up in the documentation. I will move them to the `HeaderMatcher` level to convey the right meaning in the docs

**Comment by kavyako** [comment]
@junr03 I have kept the current behavior and updated the docs. Can you please review.

File: api/envoy/api/v2/route/route.proto [hunk 1]
```diff
@@ -888,37 +888,20 @@ message RateLimit {
 //       "name": ":method",
 //       "value": "POST"
 //     }
+//
+// .. attention::
+//   In the absence of any header match specifier, match will default to :ref:`present_match <envoy_api_field_route.HeaderMatcher.present_match>`.
```
Comment (mattklein123) [inline api/envoy/api/v2/route/route.proto]: Hmm. I think we stopped running clang format on protos again as this should wrap. Let me fix and then we can merge master. Sorry for the trouble.

Comment (mattklein123) [reply api/envoy/api/v2/route/route.proto]: https://github.com/envoyproxy/envoy/pull/3811

Comment (mattklein123) [reply api/envoy/api/v2/route/route.proto]: OK please merge master and fix format. Thank you!

**Pull request merged**